### PR TITLE
AGM Task 8. Http Backend integration

### PR DESCRIPTION
**Pull request by IuriiZhuk**
- Backend integration
- add integration for authorization service
- implement load more button
- replace search behavior with server side search
- add integration for course delete/create/edit
- add interceptor to pass authorization token

### CODE REVIEW

File: src/app/core/auth/interceptors/auth-interceptor.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import { HttpRequest, HttpHandler, HttpInterceptor, HttpEvent } from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class AuthInterceptor implements HttpInterceptor {
+
+  intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+
+  const authReq = req.clone({
+  headers: req.headers.set('Authorization', 'agm-token')
```
Comment (dKab) [inline src/app/core/auth/interceptors/auth-interceptor.ts]: shouldn't you send the value that you received from BE in this header?

File: src/app/courses/pages/courses/courses.component.ts [hunk 2]
```diff
@@ -2,36 +2,55 @@ import { Component, OnInit, OnChanges } from '@angular/core';
...
-    this.courseService.deleteCourseById(id);
-    this.filteredMockCourses = this.courseService.getAllCourses();
+  public onSearchHandler(term: string): void {
+     // this.searchTerms.next(term);
+    this.courses$ = this.courseService.getCourses(this.lastCourseCount, term);
   }
 
-
+  public onLoadMoreHandler() {
+    this.lastCourseCount += 3;
```
Comment (dKab) [inline src/app/courses/pages/courses/courses.component.ts]: this should probably be a variable instead of magic number

Comment (IuriiZhuk) [reply src/app/courses/pages/courses/courses.component.ts]: fix

File: src/app/core/auth/interceptors/auth-interceptor.ts (hunk 1 above)
Comment (IuriiZhuk) [reply src/app/core/auth/interceptors/auth-interceptor.ts]: add user token

**Pull request merged**